### PR TITLE
Modification to Unhashed issue #72 created Mirrored as additional option

### PR DESCRIPTION
**Pull request by brittainmark**
I have modified the code and the docs to add a new option Mirrored that mirrors the original directory structure under bmz_cache and uses the readable file names.

This is as discussed in issue number 72. I have changed it a bit since then to use switch case for each of the three cases I have readable as the default.

I have tested on zen cart version 1.5.6c using php 7.4

Hope acceptable.

### CODE REVIEW

File: 1_Installation_Files (v1.5.5)/YOUR_ADMIN/includes/init_includes/init_image_handler.php [hunk 1]
```diff
@@ -261,7 +261,7 @@
             zen_register_admin_page('toolsImageHandlerUninstall', 'BOX_TOOLS_IMAGE_HANDLER_UNINSTALL', 'FILENAME_IMAGE_HANDLER_UNINSTALL', '', 'tools', 'N', 99);
         }
         
-        if (version_compare(IH_VERSION, '5.0.0', '<')) {
+        if (version_compare(IH_VERSION, '5.0.0', '<') || !defined('IH_CACHE_NAMING')) {
```
Comment (lat9) [inline 1_Installation_Files (v1.5.5)/YOUR_ADMIN/includes/init_includes/init_image_handler.php:264]: Why did you need to add that additional clause?

File: 1_Installation_Files (v1.5.5)/YOUR_ADMIN/includes/init_includes/init_image_handler.php [hunk 2]
```diff
@@ -272,9 +272,17 @@
...
+            $db->Execute(
+                "UPDATE " . TABLE_CONFIGURATION . "
+                     SET configuration_description = 'Choose the method that <em>Image Handler</em> uses to name the resized images in the <code>bmz_cache</code> directory.<br /><br />The <em>Hashed</em> method was used by Image Handler versions prior to 4.3.4 and uses an &quot;MD5&quot; hash to produce the filenames.  It can be &quot;difficult&quot; to visually identify the original file using this method.  If you are upgrading Image Handler from a version prior to 4.3.4 <em>and</em> you have hard-coded links in product (or other) definitions to those images, <b>do not change</b> this setting from <em>Hashed</em>.<br /><br />Image Handler v4.3.4 (unreleased) introduced the concept of a <em>Readable</em> name for those resized images.  This is a good choice for new installations of <em>IH</em> or for upgraded installations that do not have hard-coded image links.<br /><br />Image Handler v5.1.9 introduced <em>Mirrored</em> which is the same as readable but it creates a directory structure under bmz_cache that mirrors the directory structure where the original images are saved.',
```
Comment (lat9) [inline 1_Installation_Files (v1.5.5)/YOUR_ADMIN/includes/init_includes/init_image_handler.php:281]: This section should be placed towards the bottom of the initialization file, just under the update for v5.1.2, as a standalone `if` statement.  Please include lead-in commentary identifying the GitHub issue associated with the update.

File: 1_Installation_Files (v1.5.5)/YOUR_ADMIN/includes/init_includes/init_image_handler.php (hunk 1 above)
Comment (brittainmark) [reply 1_Installation_Files (v1.5.5)/YOUR_ADMIN/includes/init_includes/init_image_handler.php:264]: Trying to save an insert and an update.

File: 1_Installation_Files (v1.5.5)/YOUR_ADMIN/includes/init_includes/init_image_handler.php (hunk 2 above)
Comment (brittainmark) [reply 1_Installation_Files (v1.5.5)/YOUR_ADMIN/includes/init_includes/init_image_handler.php:281]: OK will move to make separate. Not used to git yet. Do i make the changes on my original version and then submit a new change request?

File: 1_Installation_Files (v1.5.5)/YOUR_ADMIN/includes/init_includes/init_image_handler.php (hunk 1 above)
Comment (brittainmark) [reply 1_Installation_Files (v1.5.5)/YOUR_ADMIN/includes/init_includes/init_image_handler.php:264]: Should i have left the original insert?

**Comment by brittainmark** [comment]
I think I have made the changes and they appear to be reflected in the pull request.

File: 1_Installation_Files (v1.5.5)/YOUR_ADMIN/includes/init_includes/init_image_handler.php (hunk 1 above)
Comment (lat9) [reply 1_Installation_Files (v1.5.5)/YOUR_ADMIN/includes/init_includes/init_image_handler.php]: The initialization only 'kicks in' if the version of IH is different from that previously recorded in the database, so there's no need to 'save'.

You just need to remove that added clause.

File: 1_Installation_Files (v1.5.5)/YOUR_ADMIN/includes/init_includes/init_image_handler.php (hunk 2 above)
Comment (lat9) [reply 1_Installation_Files (v1.5.5)/YOUR_ADMIN/includes/init_includes/init_image_handler.php]: You'll just update the changed version you have and re-submit the PR.

Comment (lat9) [reply 1_Installation_Files (v1.5.5)/YOUR_ADMIN/includes/init_includes/init_image_handler.php]: Thanks for doing this, BTW.  I really appreciate it!
